src/main/java/frc/robot/Subsystems/swerveModule.java

```diff
@@ -91,12 +94,34 @@ public void setTargetState(SwerveModuleState targetState, boolean isCosineCompen
             driveMotor.set(targetState.speedMetersPerSecond/Constants.attainableMaxModuleSpeedMPS); 
         }
     //FEEDBACK//
+        public Command calibrate() {
+            return runOnce(() -> {
+                SparkMaxConfig brakeConfig = new SparkMaxConfig();
+                brakeConfig.smartCurrentLimit(40);
+                brakeConfig.idleMode(IdleMode.kBrake);
+                steerMotor.configure(brakeConfig, ResetMode.kNoResetSafeParameters, PersistMode.kNoPersistParameters);
+
+                var encoderValue = moduleEncoder.getAbsolutePosition().getValueAsDouble();
+                Preferences.setDouble(EncoderPreferenceKey + encoderID, encoderValue);
+            }).withName("Calibrate");
+        }
+
+        public Command prepareToCalibrate() {
+            return runOnce(() -> {
+                SparkMaxConfig idleConfig = new SparkMaxConfig();
+                idleConfig.smartCurrentLimit(40);
+                idleConfig.idleMode(IdleMode.kCoast);
```

Comment:
@OwenM2025 You're not aware of any better way to temporary switch to coast than writing a new config? I copied in the current limit here; not sure I need to copy in the current limit, too.